plugin/db/mysql/restore.go

```diff
@@ -639,31 +634,41 @@ func (driver *Driver) FetchAllBinlogFiles(ctx context.Context, downloadLatestBin
 	return nil
 }
 
-func (driver *Driver) syncBinlogMetaFileFromCloud(ctx context.Context, client *bbs3.Client) error {
-	if client == nil {
-		return nil
+// DownloadBinlogOrMetaFileFromCloud downloads a binlog or metadata file from the cloud storage.
+// In case of network errors which will get partially downloaded files, we first download to a temporary file.
+// After that, we then rename it to the target file path.
+func (driver *Driver) DownloadBinlogOrMetaFileFromCloud(ctx context.Context, client *bbs3.Client, fileName string) error {
```

Comment:
This function should be consistent, mixing both binlog and metadata files can be troublesome.

Let's make it generic downloadFileFromCloud.

Reply:
Done.